app/components/channel_actions/manage_members_label/manage_members_label.tsx

```diff
@@ -0,0 +1,152 @@
+    role_change_error: {
+        id: t('mobile.manage_members.change_role.error'),
+        defaultMessage: 'An error occurred while trying to update the role. Please check your connection and try again.',
+    },
+    make_channel_admin: {
+        id: t('mobile.manage_members.make_channel_admin'),
+        defaultMessage: 'Make Channel Admin',
+    },
+    make_channel_member: {
+        id: t('mobile.manage_members.make_channel_member'),
+        defaultMessage: 'Make Channel Member',
+    },
+    remove_title: {
+        id: t('mobile.manage_members.remove_member'),
+        defaultMessage: 'Remove From Channel',
+    },
+    remove_message: {
+        id: t('mobile.manage_members.message'),
+        defaultMessage: 'Are you sure you want to remove the selected member from the channel?',
+    },
+    remove_cancel: {
+        id: t('mobile.manage_members.cancel'),
+        defaultMessage: 'Cancel',
+    },
+    remove_confirm: {
+        id: t('mobile.manage_members.remove'),
+        defaultMessage: 'Remove',
+    },
+});
+
+type Props = {
+    canRemoveUser: boolean;
+    channelId: string;
+    manageOption: ManageOptionsTypes;
+    testID?: string;
+    userId: string;
+}
+
+const ManageMembersLabel = ({canRemoveUser, channelId, manageOption, testID, userId}: Props) => {
+    const intl = useIntl();
+    const {formatMessage} = intl;
+    const serverUrl = useServerUrl();
+
+    const handleRemoveUser = useCallback(async () => {
+        removeMemberFromChannel(serverUrl, channelId, userId);
+        fetchChannelStats(serverUrl, channelId, false);
+        await dismissBottomSheet();
+        DeviceEventEmitter.emit(Events.REMOVE_USER_FROM_CHANNEL, userId);
+    }, [channelId, serverUrl, userId]);
+
+    const removeFromChannel = useCallback(() => {
+        Alert.alert(
+            formatMessage(messages.remove_title),
+            formatMessage(messages.remove_message),
+            [{
+                text: formatMessage(messages.remove_cancel),
+                style: 'cancel',
+            }, {
+                text: formatMessage(messages.remove_confirm),
+                style: 'destructive',
+                onPress: handleRemoveUser,
+            }], {cancelable: false},
+        );
+    }, [handleRemoveUser]);
+
+    const updateChannelMemberSchemeRole = async (schemeAdmin: boolean) => {
+        const result = await updateChannelMemberSchemeRoles(serverUrl, channelId, userId, true, schemeAdmin);
+        if (result.error) {
+            alertErrorWithFallback(intl, result.error, messages.role_change_error);
+        }
+        await dismissBottomSheet();
+        DeviceEventEmitter.emit(Events.MANAGE_USER_CHANGE_ROLE, {userId, schemeAdmin});
+    };
+
+    const onAction = useCallback(() => {
+        switch (manageOption) {
+            case REMOVE_USER:
+                removeFromChannel();
+                break;
+            case MAKE_CHANNEL_ADMIN:
+                updateChannelMemberSchemeRole(true);
```

Comment:
This is a function generated on this component, that seems to potentially change between renders.

Should we add it to the dependencies and make the `updateChannelMemberSchemeRole` to use `useCallback`, or are there reasons to keep it like this?

Reply:
suggestion implemented.